src/components/SurveyComponent.vue

```diff
@@ -1,324 +1,277 @@
-        let responseValue = getResponseValue(questionnaire, options.name, options.value);
-        // If the index is undefined, add a new entry to questionnaireResponse.item
-        if (answerItemIndex == -1) {
-          questionnaireResponse.item.push({
-            linkId: options.name,
-            answer: [{
-              [responseValue.type]: responseValue.value
-            }]
-          });
-        } else { // Otherwise update the existing index with the new response
-          questionnaireResponse.item[answerItemIndex] = {
-            linkId: options.name,
-            answer: [{
-              [responseValue.type]: responseValue.value
-            }]
-          };
-        }
-      }
-      // Need to reload the patient bundle since the responses have been updated
-      cqlWorker.postMessage({patientBundle: patientBundle});
+  mounted() {
+    this.setAuthClient().then(() => {
+      if (this.error) return; // auth error, cannot continue
+      this.setPatientId().then(() => {
+        if (this.error) return;
+        this.initializeInstrument().then(() => {
+          if (this.error) return; // error getting instrument, abort
+          this.initializeSurveyObj();
+          this.getFhirResources();
+          this.setQuestionnaireAuthor();
+          // Send the patient bundle to the CQL web worker
+          sendPatientBundle(this.patientBundle);
+          this.ready = true; // We don't show this component until `ready=true`
+        }).catch(e => {
+          this.error = e;
+          console.log("Questionnaire error ", e);
+        })
+      }).catch(e => {
+        this.error = e;
+        console.log("Patient resource error ", e);
+      });
+    }).catch(e => {
+      console.log("Auth Error ", e);
+      this.error = e;
+      this.ready = true;
     });
-    // Add a handler which will fire when the Questionnaire is submittedc
-    this.survey.onComplete.add(function() {
-      // Mark the QuestionnaireResponse as completed
-      questionnaireResponse.status = 'completed'
+  },
+  methods: {
+    initializeInstrument() {
+      var self = this;
+      return getScreeningInstrument().then(data => {
+        // Load the Questionniare, CQL ELM JSON, and value set cache which represents the alcohol screening instrument
+        const [questionnaire, elmJson, valueSetJson] = data;
+        if (!questionnaire) throw Error("No questionnaire set");
+        self.questionnaire = questionnaire;
+        // Assemble the parameters needed by the CQL
+        let cqlParameters = {
+          DisplayScreeningScores: process.env.VUE_APP_DISPLAY_SCREENING_SCORES && process.env.VUE_APP_DISPLAY_SCREENING_SCORES.toLowerCase() == "true" ? true : false,
+          QuestionnaireURL: this.getQuestionnaireURL()
+        };
+        // Send the cqlWorker an initial message containing the ELM JSON representation of the CQL expressions
+        setupExecution(elmJson, valueSetJson, cqlParameters);
```

Comment:
Just curious- is this kinda like [`executeELM()` for the AHRQR app](https://github.com/uwcirg/AHRQ-CDS-Connect-PAIN-MANAGEMENT-SUMMARY/blob/develop/src/utils/executeELM.js#L17)?

Reply:
Yeah, I believe so.